### ISSUE TITLE
[Feature request] Add code generation tooling

### ISSUE THREAD

**Issue by MonicaOlejniczak** [atlassian/tangerine#55]
## Overview
Since tangerine aims to formalize best practices, and has an opinionated file structure, we can leverage this to automatically generate files and code.

## Proposal
We can create a tangerine cli that is able to generate entire packages, components, routes, data providers, tests, etc, to facilitate a faster devloop.

Some examples:

* https://angular.io/cli/generate
* https://ember-cli.com/generators-and-blueprints

**Comment by imjeremyhi** [comment]
We have a vscode extension that aligns with tangerine best practices from a year ago. It uses yeoman to generate relevant files based on a config. Using it as a starting point should save a lot of time. This issue should be around updating our existing tool. https://github.com/atlassian/vscode-extension-jira-frontend

**Comment by nadiam84** [comment]
> This issue should be around updating our existing tool

Not necessarily, the yeoman-based generator has its limitations, it would be good to investigate whether we can have something more flexible that allows to set up and generate different types of packages and components more easily.

**Comment by lucaslago** [comment]
+1 for a cli tool, while having a vs-code extension is nice, it would be better not to be tied to an IDE since every developer has its own IDE of choice.

https://github.com/infinitered/gluegun seems an interesting and more flexible alternative to yeoman we could explore
